### PR TITLE
Allow `u32` coordinates for `textureStore`/`textureLoad`

### PR DESCRIPTION
**Pull request by PENGUINLIONG**
As illustrated in https://www.w3.org/TR/WGSL/#texturestore:

> F is a [texel format](https://www.w3.org/TR/WGSL/#texel-format)
> C is [i32](https://www.w3.org/TR/WGSL/#i32), or [u32](https://www.w3.org/TR/WGSL/#u32)
> ...

So I guess `vec<u32>` should be a valid `textureStore` coordinate.

I saw that in `back::glsl::write_texture_coord` the coord is explicitly cast to `ivec` so I think that's good. Please remind me if anything should be changed accordingly in the codegens.

### CODE REVIEW

**Review by teoxoy — CHANGES_REQUESTED** [review]
I looked over the backends and this shouldn't cause any issues.
Could you add some lines to the `tests/in/image.wgsl` file to test for those overloads?

Thanks!

**Comment by teoxoy** [comment]
This PR would in theory allow `textureLoad` to also accept `u32` coordinates. Could you also add a test for that?
Also, make sure you push the changed test snapshots (that's why CI is failing).

**Comment by teoxoy** [comment]
Looks like we need to check for `component_kind == crate::ScalarKind::Uint` and convert `array_index_i32_id` to a uint (similar to how we do for the else branch below).

https://github.com/gfx-rs/naga/blob/8e1b0529e62f39f2051ec813f99fd682d1a43b5f/src/back/spv/image.rs#L317-L336

**Comment by PENGUINLIONG** [comment]
ah thanks for the hint ;)

**Comment by PENGUINLIONG** [comment]
wow thats funny. i thought the glsl back is not affected.

**Comment by PENGUINLIONG** [comment]
Oof. Can I rerun the DXC test?

File: src/back/glsl/mod.rs [hunk 1]
```diff
@@ -3146,10 +3146,16 @@ impl<'a, W: Write> Writer<'a, W> {
             None => {
                 if tex_1d_hack {
                     write!(self.out, "ivec2(")?;
+                } else if vector_size == 1 {
+                    write!(self.out, "int(")?;
+                } else {
+                    write!(self.out, "ivec{}(", vector_size)?;
```
Comment (teoxoy) [inline src/back/glsl/mod.rs:3152]: Let's do this only if necessary (i.e. only if the `coordinate` expression's type is unsigned).

**Review by teoxoy — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**